controller/controller-common/src/main/java/com/pinterest/rocksplicator/controller/mysql/MySQLTaskQueue.java

```diff
@@ -138,28 +154,43 @@ public boolean removeCluster(final String clusterName) {
     return clusterNames;
   }
 
-  @Override
-  public boolean enqueueTask(final TaskBase taskBase,
-                             final String clusterName,
-                             final int runDelaySeconds) {
-    entityManager.getTransaction().begin();
+  private long enqueueTaskImpl(final TaskBase taskBase,
+                               final String clusterName,
+                               final int runDelaySeconds,
+                               final Task.TaskState state,
+                               final String claimedWorker,
+                               boolean inTransaction) {
```

Comment:
Or we can just make this method non-transactional and wrap it with `entityManager.getTransaction().begin()` and `entityManager.getTransaction().commit()` in the outer method.

Also for transaction, do we need to do `try {...} finally {...}`?